src/app/dynamic-forms/models/cinchy-form.model.ts

```diff
@@ -830,6 +844,21 @@ export class Form {
   }
 
 
+  /**
+   * @param rowId The rowId of the target entity
+   * @returns The index of the childFormRowValues set with data matching the given rowId, or -1 if no such row exists
+   */
+  getChildFormRowIndexByRowId(rowId: number): number {
+
+    return this.childFormRowValues?.findIndex(
+      (rowData: { [columnName: string]: any }): boolean => {
+
+        return (rowData["Cinchy ID"] === rowId);
+      }
+    ) ?? -1;
```

Comment:
```suggestion
    return this.childFormRowValues?.findIndex(
      (rowData: { [columnName: string]: any }): boolean => {

        return (rowData["Cinchy ID"] === rowId);
      }
    );
```

Reply:
'findIndex' returns -1 if index is not found. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

Reply:
That guard is for the case when childFormRowValues is not defined, in which case the function would return undefined